doc/tutorials/notifications.md

```diff
@@ -0,0 +1,235 @@
+Preface
+===
+
+**The features described in this document are not implemented yet or experimental. This document contains a preview of how developers would use the notification feature**
+
+This document explains how notifications are implemented in Elektra and how they can be used by application developers.
```

Comment:
Could you please format the text with `fmt` or at least make a linebreak for every sentence? Doing so makes the review easier to understand (which comment refers to which text).